### PR TITLE
supplemental: Extend ITs input section 4.6.2 to have examples with LI…

### PR DESCRIPTION
**Pull request by mahfouz72**
closes: #15027 
closes: #15025

### CODE REVIEW

**Comment by mahfouz72** [comment]
@romani @rdiachenko @Zopsss  Please review, the test will not pass until we have the token in our config right?

File: src/it/resources-noncompilable/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAroundWhen.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            //              ''when' is not preceded with whitespace.'
+            default -> { }
+        }
+
+        switch (o) {
+            case Integer i when (i == 0) -> { }
+            case String s when (s.equals("a")) -> { }
+            case Point(int x, int y) when (x >= 0 && y >= 0) -> { }
+            default -> { }
+        }
```
Comment (Zopsss) [inline src/it/resources-noncompilable/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAroundWhen.java:42]: can you also add test cases where `when` is not followed by brackets? something like:

```java
String day = "Monday";

switch (day) {
    case String s when s.startsWith("M"):
        System.out.println("It's a Monday!");
        break;
    case String s when s.startsWith("T"):
        System.out.println("It's a Tuesday!");
        break;
    default:
        System.out.println("It's another day.");
}
```

**Review by Zopsss — COMMENTED** [review]
item:

**Comment by nrmancuso** [comment]
I am putting this in draft until we resolve the linked issues

**Comment by Zopsss** [comment]
>  Please review, the test will not pass until we have the token in google config right?

Yes but before that we need the token support in Check itself. After that we can add it in google config

**Comment by romani** [comment]
Please format this file by CLI of google-java-format and restore only required for this Input missing space to keep violations valid.
So in this case file will look natural for good style except for required lines that are with violation by design.

**Review by nrmancuso — APPROVED** [review]

**Review by romani — APPROVED** [review]
Look ok

**Review by rnveach — APPROVED** [review]

**Pull request merged**